### PR TITLE
Report different message for FailedToCommitClusterStateException in FileSettingsService

### PR DESCRIPTION
**Pull request by prdoyle**
An alternative to #114920 for [ES-9360](https://elasticco.atlassian.net/browse/ES-9360).

This one moves the logging to an overridable error handler method.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/file/AbstractFileWatchingService.java [hunk 1]
```diff
@@ -293,12 +293,20 @@ final WatchKey enableDirectoryWatcher(WatchKey previousKey, Path settingsDir) th
...
-                listener.watchedFileChanged();
-            }
         } catch (IOException | ExecutionException e) {
-            logger.error(() -> "Error processing watched file: " + watchedFile(), e);
+            onProcessFileChangesException(e);
+            return;
         }
+        for (var listener : eventListeners) {
+            listener.watchedFileChanged();
+        }
```
Comment (prdoyle) [inline server/src/main/java/org/elasticsearch/common/file/AbstractFileWatchingService.java:302]: This loop can't throw `IOException` or `ExecutionException` anyhow. Move them out of the `try` block for clarity.

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java [hunk 2]
```diff
@@ -127,6 +128,15 @@ protected void processFileChanges() throws ExecutionException, InterruptedExcept
         completion.get();
     }
 
+    @Override
+    protected void onProcessFileChangesException(Exception e) {
+        if (e instanceof ExecutionException && e.getCause() instanceof FailedToCommitClusterStateException f) {
+            logger.error("Unable to commit cluster state", e);
```
Comment (prdoyle) [inline server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java:134]: Oh right, I forgot about the `f` here. Initially I was just logging `f` but decided that more context never hurts so I just logged the original exception `e`.

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java:150]: You're still creating a variable `f` which is not referenced

**Comment by prdoyle** [comment]
Backport in #134909.

**Pull request merged**